tally_ho/apps/tally/static/js/barcode_verify.js

```diff
@@ -1,27 +1,68 @@
 function barcodes_match(empty_message, length_message, mismatch_message) {
-    var barcode = document.getElementById('id_barcode');
-    var barcode_copy = document.getElementById('id_barcode_copy');
+    const barcode = document.getElementById('id_barcode');
+    const barcode_copy = document.getElementById('id_barcode_copy');
+    const barcode_scan_entry = document.getElementById('barcode_scan_entry').hidden;
+    if (barcode_scan_entry){
+        if (barcode.value == barcode_copy.value) {
+            barcode_copy.parentNode.setAttribute('class', 'has-success');
+            return true;
+        }
+        barcode_copy.parentNode.setAttribute('class', 'has-error');
 
-    if (barcode.value == barcode_copy.value) {
-        barcode_copy.parentNode.setAttribute('class', 'has-success');
-        return true;
-    }
-    barcode_copy.parentNode.setAttribute('class', 'has-error');
-
-    if (barcode.value == "" || barcode_copy.value == "") {
-        barcode.parentNode.setAttribute('class', 'has-error');
-        alert(empty_message);
-    } else if (barcode.value.length < 1) {
-        barcode.parentNode.setAttribute('class', 'has-error');
-        alert(length_message);
+        if (barcode.value == "" || barcode_copy.value == "") {
+            barcode.parentNode.setAttribute('class', 'has-error');
+            alert(empty_message);
+        } else if (barcode.value.length < 1) {
+            barcode.parentNode.setAttribute('class', 'has-error');
+            alert(length_message);
+        }
+        else {
+            alert(mismatch_message);
+        }
     }
-    else {
-        alert(mismatch_message);
+
+    return;
+}
+
+let barcode_number = '';
+
+const barcode_scanner = (barcode_input_field) => {
+    if (barcode_input_field) {
+        barcode_input_field.addEventListener('keydown', function (e) {
+            e.preventDefault();
+            let textInput = e.key
+            if (textInput.length === 1 || (e.keyCode === 13)) {
+                if (!(isNaN(parseInt(textInput))) && e.code == 'KeyA') {
+                    barcode_number = barcode_number + textInput;
```

Comment:
I'm unclear on why this is adding to `barcode_number` as opposed to simply assigning `textInput`

Reply:
The barcode scanner is keying in single keys hence the need to concatenate them all and generate the full barcode.